docs: add Amount decisions (#993)
razorpay/blade: packages/blade/src/components/Amount/_decisions/decisions.md

# Amount <!-- omit in toc -->

Amount component is a UI element that displays and formats various currency values in a specified format.

<img  src="./amount-thumbnail.png"  width="380" />

## Design

- [Figma - Amount](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=28012%3A580578&t=3peAz8A2n2Gw4WMl-1)

## API

| Prop           | Type                                                               | Default     | Description                                                                    | Required |
| -------------- | ------------------------------------------------------------------ | ----------- | ------------------------------------------------------------------------------ | -------- |

> | children       | `string`                                                           | `undefined` | The value to be rendered within the component                                  | ✅       |

Reviewer: With amount, we're dealing with numbers. I think the value being passed should be a number otherwise it'll involve internal parsing of string to number. Instead we can have a `value` prop for accepting the numeric literal.

> | children                  | `number`                                                   | `undefined` | The value to be rendered within the component                                  | ✅       |

Author: I was thinking of following a similar approach as other components and taking the value as a string in children and then manipulating it accordingly. We can simply check it if the string is a valid number.

> | children                  | `number`                                                   | `undefined` | The value to be rendered within the component                                  | ✅       |

Reviewer: Added a comment [below](https://github.com/razorpay/blade/pull/993#discussion_r1108115833)

> | intent                    | `positive`, `negative`, `notice`, `information`, `blue`    | `undefined` | The variant of the amount to be rendered                                       |          |

Reviewer: Is this `intent` or `variant`? We also have `blue`

Author: `variant` is more apt here.

| fontWeight                | `bold`, `regular`                                          | `bold`      | The weight of the amount to be rendered                                        |          |

> | suffix                    | `true`, `false`                                            | `false`     | Indicates whether a suffix should be used                                      |          |

Reviewer: What about the `showCurrencySuffix` from earlier? Just `suffix` sounds vague.

Author: changed it to `hasSuffix`.

Reviewer: Can we use `showSuffix`? Generally we're following `show*` terminology in other components.

Author: cool, done.

> | hasDecimals    | `true`, `false`                                                    | `true`      | Indicates whether the amount has or can have decimals                          |          |

Reviewer: Is there a usecase for showing no. of decimals? For example decimal upto 1 place / 2 place etc.? Should be confirmed with design.

Author: @saurav12 What is the required spec here? and what should be the logic of rounding here?

Reviewer: Please also confirm on figma

> | hasTextPostfix | `true`, `false`                                                    | `true`      | Indicates whether a text postfix should be used                                |          |

Reviewer: We're generally following `show*` terminology across other components. Think this should also follow similar naming.

> | hasTextPostfix | `true`, `false`                                                    | `true`      | Indicates whether a text postfix should be used                                |          |

Reviewer: Can there be both text postfix and decimals? If not I think both these props should be clubbed in a way that doesn't allow impossible combinations.

Author: It is not impossible to have both postfix and decimal, we can have 21.20 l as a value.

Reviewer: Is this case documented on Figma 🤔

Reviewer: I couldn't find this particular case on figma, where it shows both decimals and a suffix like `100.00 k`

Reviewer: Wondering if we can consolidate decimals and suffix usecases in this case.

Author: @divyanshu013 I think we are settling at using `hasSuffix` as the only prop for both decimals and humanize. Looking at design we will not be supporting both decimal and suffix at the same time.

Reviewer: Added a comment [below](https://github.com/razorpay/blade/pull/993#discussion_r1108096596)

> | hasTextPostfix | `true`, `false`                                                    | `true`      | Indicates whether a text postfix should be used                                |          |

Reviewer: text postfix seems like a very vague terminology. As a dev I'm not able to visualize what this would do? I think we should use a clearer terminology. Same can be communicated to design and updated in figma.

Author: changed it to **showCurrencyPostfix**

Reviewer: Sounds good, can you please communicate and get that updated on figma too?

> | hasTextPostfix | `true`, `false`                                                    | `true`      | Indicates whether a text postfix should be used                                |          |

Reviewer 2: Yup it should be `showCurrencyPostfix`

> 

Reviewer: Don't see the APIs for decimals 🤔

Author: `hasSuffix` will have three states `Decimals`, `Humanize` and `None`. Please check figma once.

Reviewer: But in API decision table I only see `true` and `false`

Author: Opps, my bad. This got updated later, I did not update the prop values. It's updated now.

> | size           | `3xl`, `2xl`, `xl`, `large`, `medium`, `small`                     | `3xl`       | The size of the amount to be rendered                                          |          |

Reviewer 2: We should prefer not to use shortforms in the API. Also, do we need so many sizes?
```suggestion
| size           | `3xlarge`, `2xlarge`, `xlarge`, `large`, `medium`, `small`                     | `3xlarge`       | The size of the amount to be rendered                                          |          |
```

### Sample Usage

```jsx
import { Amount } from '@razorpay/components';

<Amount
  size="2xlarge"
  isSuffixPrefixHighlighted={true}
  fontWeight="regular"
  suffix={false}
  intent="information"
>

>   1000.27

Reviewer: Will have to be `{1000.27}` with proposed API. Added a comment to consider a `value` prop.

Author: I think we should use the same approach as other components.

Reviewer: Linking to my reasonings [above](https://github.com/razorpay/blade/pull/993#discussion_r1099986273). I feel the same approach wouldn't work here as well because this component is specifically designed to work with numbers and currencies.

Author: Sure, let's do that.

> | isStyled       | `true`, `false`                                                    | `true`      | Highlight the main amount by making the prefix symbol and decimal digits small |          |

Reviewer: Similarly, this also reads as a vague terminology. styled doesn't actually convey the intent of the prop.

Author: @chaitanyadeorukhkar @divyanshu013 Can you folks suggest a better name here.

Reviewer: Maybe `isAmountEnlarged`, `isAmountHighlighted` though a bit verbose prop

Author: `isHighlighted` then?

Reviewer: Can't think of a better prop name right now 🤔 

Definitely seems better than `isStyled`.

Author: @divyanshu013 @kamleshchandnani suggested to use `isSuffixPrefixHighlighted` instead.

> | screenSize     | `desktop`, `mobile`                                                | `desktop`   | Provides the size of the component based on the platform                       |          |

Reviewer 2: This is for design only. On code we automatically handle this based on our platform specific tokens. So if you choose `font[100]` it will reflect to a different size on web & mobile. 
```suggestion
```

>   isStyled="true"

Reviewer 2: This should be boolean
```suggestion
  isStyled={true}
```

  hasTextPostfix="true"
  intent="information"
  contrast="high"
>
  1000.27

> </Amount>;

Reviewer: Should be updated to use booleans as `{false}` etc.

```

## Open Questions

#### Dev

> - Should we call it `variant` or `intent`, `isStyled` or something more apt?

Reviewer: In other components we're using `intent` where they represent a feedback. In others such as badge where there are non semantic naming like "blue" we're using variant.

Author: changed it to **variant**

> - Should we separate the amount with commas based on the Indian currency format?

Reviewer 2: Are we only supporting 1 currency format (INR)? Or should we allow the consumer to decide what format it should be?

Reviewer 3: For now, we will be supporting only INR. Going forward we will add more currencies. One of the major problems right now is we use ₹ and INR interchangeably. But in Malaysia we only have RM and there is no symbol associated with it. Hence we are staying away from this. Will solve this once the use-case arrives.

Reviewer 4: Just leaving my thoughts here. 
Dashboard is currently preparing for launching in Malaysia. I am not sure about the timelines but I do see a big chunk of code already being merged to prepare for it, so the use-case might arrive sooner. Perhaps we can also support Malaysia along with INR for now.

for reference this is one of the PR's which was merged couple of months back
https://github.com/razorpay/dashboard/pull/11454/files

Reviewer 2: Agreed. We should start thinking about currency support from now. We can limit it to just INR for now and add more later

> - Should we separate the amount with commas based on the Indian currency format?

Reviewer: On the designs I see this is being done to all values. In this case we should do this as a default. If there are usecases for values without commas could you confirm with design if they're needed?

Author: @saurav12 Can you check this?

Reviewer: Please also add comments on figma at the occurence for design

Author: Done.

Reviewer 5: is this fixed on design @Priyanshu-C? what is the conclusion?

- What should be the default `intent`? Kept it as `default` for now, but `neutral` seems more apt.

> - Scope of A11y?

Reviewer 2: Any update on this @Priyanshu-C ^

> - Scope of A11y?

Reviewer: Let's do some exploration on this too please.

#### Design

> - Do we need to support more currency types?

Reviewer 2: +1

@saurav12

## API

| Prop                      | Type                                                       | Default     | Description                                                                    | Required |
| ------------------------- | ---------------------------------------------------------- | ----------- | ------------------------------------------------------------------------------ | -------- |
| children                  | `number`                                                   | `undefined` | The value to be rendered within the component                                  | ✅       |
| intent                    | `positive`, `negative`, `notice`, `information`, `blue`    | `undefined` | The variant of the amount to be rendered                                       |          |
| fontWeight                | `bold`, `regular`                                          | `bold`      | The weight of the amount to be rendered                                        |          |
| hasDecimals               | `true`, `false`                                            | `true`      | Indicates whether the amount has or can have decimals                          |          |

> | suffix                    | `true`, `false`                                            | `false`     | Indicates whether a text suffix should be used                                 |          |

Reviewer 2: ```suggestion
| hasSuffix                    | `true`, `false`                                            | `false`     | Indicates whether a text suffix should be used                                 |          |
```

| size                      | `3xlarge`, `2xlarge`, `xlarge`, `large`, `medium`, `small` | `3xlarge`   | The size of the amount to be rendered                                          |          |

> | isSuffixPrefixHighlighted | `true`, `false`                                            | `true`      | Highlight the main amount by making the prefix symbol and decimal digits small |          |

Reviewer 2: Where is the prefix?

Author: They are referring to the Rupee symbol here.

Reviewer: Should we just make this `isHighlighted`, seems too verbose right now and actually we're not highlighting them, instead we're shrinking them 😅

Author: That does makes sense and it `isHighlighted` is not too verbose, @kamleshchandnani @chaitanyadeorukhkar @divyanshu013 final call.

Reviewer 2: How about `isAffixHighlighted`?

Prefixes and Suffixes are types of affixes. This way we're not being too vague about what it means too.

Author: Sure, this works.

import { Amount } from '@razorpay/components';

<Amount
  size="2xlarge"
  isSuffixPrefixHighlighted={true}
  fontWeight="regular"
  hasDecimals={true}
  suffix={false}
  intent="information"
>
  1000.27
</Amount>;
```

## Open Questions

#### Dev

- Should we call it `variant` or `intent`, `isStyled` or something more apt?
- Should we separate the amount with commas based on the Indian currency format?
- What should be the default `intent`? Kept it as `default` for now, but `neutral` seems more apt.
- We currently use amount value in paise in backend, so how should we convert it to rupee here?
- Scope of A11y?

#### Design

- Do we need to support more currency types?

## Discussions

### Should we separate the amount with commas based on the Indian currency format?

- For now, we will be supporting only INR. Going forward we will add more currencies. One of the major problems right now is we use ₹ and INR interchangeably. But in Malaysia we only have RM and there is no symbol associated with it. Hence we are staying away from this. Will solve this once the use-case arrives.

## References

- [Ant](https://ant.design/components/input)
- [Primeng](https://primeng.org/inputnumber)

> - [Blueprint](https://blueprintjs.com/docs/#core/components/file-input)

Reviewer 2: These seem unrelated?

<img  src="./amount-thumbnail.png"  width="380" />

## Design

- [Figma - Amount](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=28012%3A580578&t=3peAz8A2n2Gw4WMl-1)

## API

| Prop                | Type                                                       | Default     | Description                                                                    | Required |
| ------------------- | ---------------------------------------------------------- | ----------- | ------------------------------------------------------------------------------ | -------- |
| children            | `number`                                                   | `undefined` | The value to be rendered within the component                                  | ✅       |
| variant             | `positive`, `negative`, `notice`, `information`, `blue`    | `undefined` | The variant of the amount to be rendered                                       |          |
| contrast            | `low`, `high`                                              | `low`       | The contrast of the amount to be rendered                                      |          |

> | fontWeight          | `bold`, `regular`                                          | `bold`      | The weight of the amount to be rendered                                        |          |

Reviewer 6: 🤔 should we add the `font` prefix here? In Text we are naming it just `weight` 

/cc @saurav12

Reviewer: `fontWeight` definitely is clearer to understand as a prop

| hasDecimals         | `true`, `false`                                            | `true`      | Indicates whether the amount has or can have decimals                          |          |

> | showCurrencyPostfix | `true`, `false`                                            | `false`     | Indicates whether a text postfix should be used                                |          |

Author: Done.

| value                     | `number`                                                   | `undefined` | The value to be rendered within the component                                                                                                                                                     | ✅       |
| variant                   | `positive`, `negative`, `notice`, `information`, `blue`    | `undefined` | The variant of the amount to be rendered                                                                                                                                                          |          |
| fontWeight                | `bold`, `regular`                                          | `bold`      | The weight of the amount to be rendered                                                                                                                                                           |          |

> | showSuffix                | `none`, `decimals` , `humanise`                            | `decimals`  | The type of affix to be displayed with the currency value, (none, decimals, and humanise) that respectively indicate no affix, the number of decimal places, and the scale of the currency value. |          |

Reviewer: Since this isn't a boolean anymore we can avoid the `show` prefix and just make the prop `suffix` perhaps:

```ts
suffix: 'none' | 'decimals' | 'humanise'
```

One thing to confirm whether to opt for `humanise` (british) or `humanize` (american) 😸

Reviewer 2: +1

I'd vote for `humanize`

| size                      | `3xlarge`, `2xlarge`, `xlarge`, `large`, `medium`, `small` | `3xlarge`   | The size of the amount to be rendered                                                                                                                                                             |          |

> | isSuffixPrefixHighlighted | `true`, `false`                                            | `true`      | Highlight the main amount by making the prefix symbol and decimal digits small                                                                                                                    |          |

Reviewer: Are we not opting for affix? /cc @chaitanyadeorukhkar

Author: This is still open for discussion, not sure what we are finalizing.

> | isSuffixPrefixHighlighted | `true`, `false`                                            | `true`      | Highlight the main amount by making the prefix symbol and decimal digits small                                                                                                                    |          |

Reviewer 2: We need to changes this to:
```suggestion
| isAffixHighlighted | `true`, `false`                                            | `true`      | Highlight the main amount by making the prefix symbol and decimal digits small                                                                                                                    |          |
```

Author: @chaitanyadeorukhkar when this value will be true, the affix won't be highlighted, it is the value which will get highlighted.

Author: Yes, it is incorrect and open to be replaced with a better name.

Author: **isHighlighted=true** 
<img width="503" alt="image" src="https://user-images.githubusercontent.com/63398163/220062152-dd4c7351-e062-4dd0-9594-26302f378175.png">
**isHighlighted=false** 
<img width="503" alt="image" src="https://user-images.githubusercontent.com/63398163/220062244-a5d1ce15-f222-4714-b5a6-4aac23f635a4.png">

Reviewer 2: Can it be like this:

`isAffixHighlighted=true`
![](https://user-images.githubusercontent.com/63398163/220062244-a5d1ce15-f222-4714-b5a6-4aac23f635a4.png)

`isAffixHighlighted=false`
![](https://user-images.githubusercontent.com/63398163/220062152-dd4c7351-e062-4dd0-9594-26302f378175.png)

Reviewer: This is really taking a test of my English grammar 😅 

How about `isAffixDeemphasized`?

Reviewer 2: Why go in the negation though? `isAffixHighlighted` would still work

Reviewer: Because it isn't actually "highlighted" (in terms of the language), it matches the other content (the number part - both have same size, weight etc.) 😅

Reviewer 2: Yeah, Makes sense. How about `isAffixSubtle`? Seems a bit simpler to write than `isAffixDeemphasized`

Reviewer 7: Should we call it `isCurrencyHighlighted` instead? I understand that it also changes emphasis on decimal number but maybe just to keep the name simple?

TIL `Affix` 🙈

Reviewer: Actually currency isn't "highlighted" per se, just the other things are shrinked and faded 😅 

So far these two seems like best options to me:
- `isAffixDeemphasized`
- `isAffixSubtle`

Author: let's go with this `isAffixSubtle` ?

Reviewer 7: +1 to `isAffixSubtle`

Reviewer 2: Lets fix `isAffixSubtle` then 👍

### Sample Usage

```jsx
import { Amount } from '@razorpay/components';

<Amount

>   showSuffix="decimals"

Reviewer: To be updated based on prop name change.

  size="2xlarge"
  isSuffixPrefixHighlighted={true}
  fontWeight="regular"
  variant="information"
  value={1000}
/>;
```

> ## Open Questions

Reviewer: Let's also add the final resolutions for the open questions now.

# Amount <!-- omit in toc -->

Amount component is a UI element that displays and formats various currency values in a specified format.

<img  src="./amount-thumbnail.png"  width="380" />

## Design

- [Figma - Amount](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=28012%3A580578&t=3peAz8A2n2Gw4WMl-1)

> ## API

Reviewer: Let's also get the props synced to figma. Could you please followup with @saurav12 and ensure these are updated on the component figma?

| Prop          | Type                                                       | Default     | Description                                                                                                                                                                                       | Required |
| ------------- | ---------------------------------------------------------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
| value         | `number`                                                   | `undefined` | The value to be rendered within the component                                                                                                                                                     | ✅       |

> | variant       | `positive`, `negative`, `notice`, `information`, `blue`    | `undefined` | The variant of the amount to be rendered                                                                                                                                                          |          |

Reviewer 5: do we have blue color for Amount? can you check on designer once? have mentioned you in a comment on Figma

Author: I am not sure about the requirement, I thought we will add it the same way we did for Badge.

> | fontWeight    | `bold`, `regular`                                          | `bold`      | The weight of the amount to be rendered                                                                                                                                                           |          |

Reviewer 5: ```suggestion
| weight    | `bold`, `regular`                                          | `bold`      | The weight of the amount to be rendered                                                                                                                                                           |          |
```

we have been using weight across Blade for fontWeight

> | suffix        | `none`, `decimals` , `humanize`                            | `decimals`  | The type of affix to be displayed with the currency value, (none, decimals, and humanize) that respectively indicate no affix, the number of decimal places, and the scale of the currency value. |          |

Reviewer 5: ```suggestion
| suffix        | `none`, `decimals`, `humanize`                            | `decimals`  | The type of suffix to be displayed with the currency value, (none, decimals, and humanize) that respectively indicate no suffix, the number of decimal places, and the scale of the currency value. |          |
```
affix includes suffix and prefix both

| size          | `3xlarge`, `2xlarge`, `xlarge`, `large`, `medium`, `small` | `3xlarge`   | The size of the amount to be rendered                                                                                                                                                             |          |

> | isAffixSubtle | `true`, `false`                                            | `true`      | Highlight the main amount by making the prefix symbol and decimal digits small and faded                                                                                                          |          |

Reviewer 5: ```suggestion
| isAffixSubtle | `true`, `false`                                            | `true`      | Makes the prefix symbol and decimal digits small and faded                                                                                                          |          |
```

Highlighted and faded sounds contradicting

Reviewer 5: also, suffix is not changed when the suffix type is humanise the color remains the same

### Sample Usage

```jsx
import { Amount } from '@razorpay/components';

<Amount
  suffix="decimals"
  size="2xlarge"
  isAffixSubtle={true}
  fontWeight="regular"
  variant="information"
  value={1000}
/>;
```

> ## Open Questions

Reviewer: Let's also add resolutions to the open questions alongside.

> - Should we call `isSuffixPrefixHighlighted` or `isHighlighted` or `isAffixHighligted`, considering when this value is true the value will be highligted and not the affix.

Reviewer 5: `isAffixSubtle` is what we agreed upon so lets add that conclusion here?

> #### Dev

Reviewer 5: @Priyanshu-C lets add all the answers to the questions based on the discussions inline so we can then focus on closing the remaining ones?

- Should we call it `variant` or `intent`, `isStyled` or something more apt?
- Should we separate the amount with commas based on the Indian currency format?
- What should be the default `intent`? Kept it as `default` for now, but `neutral` seems more apt.

> - We currently use amount value in paise in backend, so how should we convert it to rupee here?

Reviewer: How are these values generally sent? I thought it might be in rupees for example `0.01` INR?

Author: Currently in X, they use it in rupee and the value transformation is handled by the consumer.

# Amount <!-- omit in toc -->

> A amount component is a UI element that displays and formal various currency values in a specified format.

Reviewer: ```suggestion
Amount component is a UI element that displays and formats various currency values to a specified format.
```

<img  src="./amount-thumbnail.png"  width="380" />

## Design

- [Figma - Amount](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=28012%3A580578&t=3peAz8A2n2Gw4WMl-1)

## API

> | Prop | Type | Default | Description | Required |

Reviewer: Table formatting is broken, please validate in markdown preview.

Author: Fixed.

|---|---|---|---|---|

| children | `string` | `undefined` | The label to be rendered within the badge. | ✅ |

| intent | `default`, `positive`,`negative`, `notice`, `information`, `blue` | `default` | The variant of the badge to be rendered. | |

> | contrast | `low`, `high` | `low` The contrast of the badge to be rendered. | |

Reviewer: Seems stale, mentions "badge"

Author: Fixed.

| screenSize | `Desktop`, `Movbile` The font weight of the badge's font. | |

### Sample Usage

```jsx
import { Amount } from '@razorpay/components';

<Amount
  size="2XL"
  isStyled="true"
  fontWeight="regular"
  hasDecimals="true"
  hasTextPostfix="true"
  intent="information"
  contrast="high"
>
  1000.27
</Amount>;
```

## Open Questions

#### Dev

- Should we call it `variant` or `intent`?

- Should we separate the amount with commas based on the Indian currency format?

- What should be the default `intent`? Kept it as `default` for now, but `neutral` seems more apt.

- Scope of A11y?

#### Design

- Do we need to support more currency types?

## Discussions

## References

- [Ant](https://ant.design/components/input)

- [Primeng](https://primeng.org/inputnumber)

- [Blueprint](https://blueprintjs.com/docs/#core/components/file-input)

- [Backbase](https://designsystem.backbase.com/v1/components/amount/web)

- [Mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat)

- [Salesforce](https://developer.salesforce.com/docs/component-library/bundle/ui:inputCurrency)

- [Dribbble](https://dribbble.com/tags/money_components)

> - [Pega](https://design.pega.com/design/currency/)

Reviewer: Formatting issue, remove line breaks.

Author: Fixed.

# Amount <!-- omit in toc -->

Amount component is a UI element that displays and formats various currency values in a specified format.

<img  src="./amount-thumbnail.png"  width="380" />

## Design

- [Figma - Amount](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=28012%3A580578&t=3peAz8A2n2Gw4WMl-1)

## API

> | Prop          | Type                                                                                                                                                                              | Default           | Description                                                                                                                                                                                         | Required |

Author: Currency, i am hiding this prop, although implementation for both INR and MYR have been done. We can enable it later.

> | Prop          | Type                                                                                                                                                                              | Default           | Description                                                                                                                                                                                         | Required |

Reviewer 5: we can expose the prop with default value of INR?

| ------------- | ---------------------------------------------------------------------------------------------- | ----------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
| value         | `number`                                                                                       | `undefined`       | The value to be rendered within the component                                                                                                                                                       | ✅       |

> | intent        | `positive`, `negative`, `notice`, `information`                                                | `undefined`       | The intent of the amount to be rendered                                                                                                                                                             |

Reviewer 2: There's also `neutral` right?

> | intent        | `positive`, `negative`, `notice`, `information` ,`neutral`                                                                                                                        | `undefined`       | The intent of the amount to be rendered                                                                                                                                                             |

Reviewer 5: i don't see `neutral` on figma. but i see `none` . can you once confirm @chaitanyadeorukhkar @Priyanshu-C what is finalised?

Author: Removed.

> | prefix        | `currency-symbol`, `currency-code`                                                                                                                                                | `currency-symbol` | The type of prefix to be shown                                                                                                                                                                      |          |

Reviewer 5: the values shall be title case? `currencyCode`, `currencySymbol`?

Author: I check List decisions, we have used 'ordered-filled'. I was thinking to use the same kebab case.

- After careful consideration, we decided to name this property `isAffixSubtle`. This decision was reached based on the fact that the property controls the behavior of the affix, which in turn affects the visual appearance of the prefix and decimal digits. By setting this property to 'true', the affix will appear smaller and more faded.

### Should we have a seperate prop for decimals and another for affix symbols?

- We initially considered using two different props, but then realized that only one of them would be shown at a time. So in that case, we decided to use a single prop called 'suffix'. We ultimately settled on using a prop with three possible values: 'none', 'decimals', and 'humanize'.

### Should we call `weight` or `fontWeight`?

- We decided to call it `weight` as this represents the weight of the amount component with the affix changing accordingly. This also follows the same convention as other components.
- Eventually we removed this prop as size prop will give weight too

### Should we separate the amount with commas based on the Indian currency format?

> - For now, we will be supporting only INR. Going forward we will add more currencies. One of the major problems right now is we use ₹ and INR interchangeably. But in Malaysia we only have RM and there is no symbol associated with it. Hence we are staying away from this. Will solve this once the use-case arrives.

Author: Using the same, discussion was not updated.
